src/Microsoft.Azure.WebJobs.Host/Config/JobHostRetryOptions.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Configuration options for controlling function execution retry behavior.
+    /// </summary>
+    public class JobHostRetryOptions
+    {
+        /// <summary>
+        /// The retry count.
+        /// </summary>
+        public int RetryCount { get; set; }
+
+        /// <summary>
+        /// The duration to wait for for a particular retry attemp
+        /// </summary>
+        public TimeSpan SleepDuration { get; set; } = TimeSpan.Zero;
+
+        /// <summary>
+        /// Gets a value indicating whether the expbnential backoff is used.
+        /// </summary>
+        public bool ExponentialBackoff { get; set; }
```

Comment:
It's confusing to have both linear and exponential stategies encoded in this single options. E.g. what if Exponetial is true and a Delay is also specified? The delay doesn't apply.

I wonder if instead we shouldn't have an options hierarchy:

 ```csharp
JobHostRetryOptions
{
    int MaxRetries
}

JobHostLinearBackoffRetryOptions : JobHostRetryOptions
{
    TimeSpan Delay
}

JobHostExponentialBackoffRetryOptions : JobHostRetryOptions
{
    TimeSpan MinimumDelay
    TimeSpan MaximumDelay
}

```

For the exponential, I propose a min/max interval based on how our existing RandomizedExponentialBackoffStrategy works.

This would then translate to also having a hierarchy in the attributes: RetryAttribute, LinearBackoffRetryAttribute, ExponentialBackoffRetryAttribute.

Reply:
Added this. Please take a look.